track.go

```diff
@@ -209,21 +209,30 @@ func (track *baseTrack) bind(ctx webrtc.TrackLocalContext, specializedTrack Trac
 }
 
 func (track *baseTrack) unbind(ctx webrtc.TrackLocalContext) error {
-	track.mu.Lock()
-	defer track.mu.Unlock()
-
-	ch, ok := track.activePeerConnections[ctx.ID()]
-	if !ok {
-		return errNotFoundPeerConnection
+	ch, err := track.removeActivePeerConnection(ctx.ID())
+	if err != err {
+		return err
 	}
 
 	doneCh := make(chan struct{})
 	ch <- doneCh
 	<-doneCh
```

Comment:
I don't understand the use of doneCh and getting from empty doneCh.

Reply:
I am finding it difficult to understand that won't line 221 creates an error like `fatal error: all goroutines are asleep - deadlock!`? And line 222 will wait for forever since there is nothing in doneCh which will create deadlock situation.

Reply:
https://play.golang.org/p/QufENXCZpO-

Reply:
https://play.golang.org/p/qBdpOKwdorR
as written in https://github.com/pion/mediadevices/pull/337#discussion_r739964814

Reply:
Nice. This is a cool trick. Thanks @at-wat , lesson learned. 😃